core/drivers/crypto/caam/utils/utils_dmaobj.c

```diff
@@ -1394,9 +1394,9 @@ TEE_Result caam_dmaobj_sgtbuf_build(struct caamdmaobj *obj, size_t *length,
 		obj->sgtbuf.paddr = obj->sgtbuf.buf->paddr;
 	}
 
-	*length = max_length;
+	*length = obj->sgtbuf.length;
```

Comment:
This is the major fix explained by the usecase explain in the commit message. 

The caam_cipher_mac.c fix is a consequence of util_dmaobj.c function fix.

Reply:
That wasn't clear to me. Some fixes are easier to understand if you explain the situation before the patch and then how the patch fixes that.

Would it make sense to include a:
`Fixes: 38923d487567 ("drivers: caam: implement CAAM DMA Object")`
line in the commit?

Do you need to update the description of the `caam_dmaobj_sgtbuf_build()`

Reply:
I reword the commit message to make it clearer (hopefully).
I reviewed the description of the `caam_dmaobj_sgtbuf_build()` function that was no more correct because the DMA buffer is not allocated in this function.
I added the commit id fixed in the commit message too.